prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,171 @@
+	repo := pre.Repo.Name
+	wrongOwnersFiles := map[string]error{}
+
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(org, repo, pre.Number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %v", err)
+	}
+
+	// List modified OWNERS files.
+	var modifiedOwnersFiles []string
+	for _, change := range changes {
+		if filepath.Base(change.Filename) == ownersFileName {
+			modifiedOwnersFiles = append(modifiedOwnersFiles, change.Filename)
+		}
+	}
+	if len(modifiedOwnersFiles) == 0 {
+		return nil
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.Clone(pre.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		path := filepath.Join(r.Dir, f)
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to read %s.", path)
+			return nil
+		}
+		var approvers []string
+		var labels []string
+		simple, err := repoowners.ParseSimpleConfig(b)
+		if err != nil || simple.Empty() {
+			full, err := repoowners.ParseFullConfig(b)
+			if err != nil {
+				wrongOwnersFiles[f] = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", f, err)
+				continue
+			} else {
+				// it's a FullConfig
+				for _, config := range full.Filters {
+					approvers = append(approvers, config.Approvers...)
+					labels = append(labels, config.Labels...)
+				}
+			}
+		} else {
+			// it's a SimpleConfig
+			approvers = simple.Config.Approvers
+			labels = simple.Config.Labels
+		}
+		// Check labels against blacklist
+		if sets.NewString(labels...).HasAny(labelsBlackList...) {
+			wrongOwnersFiles[f] = fmt.Errorf("OWNERS file contains blacklisted labels: %s", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)).List())
+			continue
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(approvers) == 0 {
+			wrongOwnersFiles[f] = errors.New("no approvers in the root directory OWNERS file")
+			continue
+		}
+	}
+	// React if we saw something.
+	if len(wrongOwnersFiles) > 0 {
+		if err := ghc.AddLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			return err
+		}
+		log.Debugf("Creating a review for %d OWNERS files.", len(wrongOwnersFiles))
+		var comments []github.DraftReviewComment
+		for errFile, errMsg := range wrongOwnersFiles {
+			resp := fmt.Sprintf("Invalid OWNERS file: %v", errMsg)
+			comments = append(comments, github.DraftReviewComment{
+				Path: errFile,
+				Body: resp,
+			})
+		}
+		err := ghc.CreateReview(org, repo, pre.Number, github.DraftReview{
+			Action:   github.Comment,
+			Comments: comments,
+		})
+		if err != nil {
+			return fmt.Errorf("error creating a review for invalid OWNERS file(s): %v", err)
+		}
+	}
```

Comment:
We also need to handle the case in which there aren't any invalid OWNERS files by removing the `invalidOwnersLabel` if it exists. Otherwise contributors would have to scrap their PR and make a new one if they accidentally invalidate an OWNERS file.
You can use the same pattern used here: https://github.com/kubernetes/test-infra/blob/982c128a275ca294967531310bf0c6653b2f1c00/prow/plugins/lgtm/lgtm.go#L212-L219